docs/content/en/docs/concepts/executable-playbooks.md

```diff
@@ -0,0 +1,387 @@
+
+Example of a `start` step:
+
+```json
+"start--d6c44626-c9b6-426b-ad5d-3311bafaf068": {
+    "on_completion": "action--4b08af84-3741-48ca-8c92-df1557a87379",
+    "type": "start"
+}
+```
+
+According to the CACAO specification, every branch of steps should end in a (unique) `end` step. This is the only step type may not specify a next step:
+
+```json
+"end--60fc8d0c-3677-4363-8576-9ea9014f8c8e": {
+    "type": "end"
+}
+```
+
+### If-condition, while-condition, and parallel steps
+
+An `if-condition` step allows executing different branches depending on a specified condition. The step must specify an `on_true` field, which references the start of a branch of steps that should be executed if the condition evaluates to true. Optionally, the `if-condition` step can define an `on_false` field that defines an alternative branch that is executed if the condition evaluates to false. In each case, the specified branch keeps executing until it encounters an `end` step.
+
+The `condition` field contains a string that specifies a [STIX Pattern](https://docs.oasis-open.org/cti/stix/v2.0/stix-v2.0-part5-stix-patterning.html). Currently, SOARCA only supports a very small subsection of the STIX Patterning specification. We support string based equality (`'a' = 'a'`) and inequality (`'a' != 'b'`) comparison. Example:
+
+```json
+"if-condition--4b95eaa4-944a-4a9d-88d4-1374a70dbacd": {
+    "name": "If it is not new years",
+    "description": "Checks if it is 01-01-2025",
+    "on_completion": "end--db937fc8-3a42-41cc-b828-ec2db212f425",
+    "type": "if-condition",
+    "condition": "__soarca_ssh_result__:value != '01-01-2025'",
+    "on_true": "action--7fe08053-3685-4d8c-bc0a-40efce75113e"
+}
+```
+
+SOARCA supports variable interpolation, which means that variables can be used inside the `condition` field, as seen in the example above.
+
+Similarly, CACAO specifies `while-condition` steps, whose `on_true` branch will be repeatedly executed until the condition evaluates to false. The execution limit is currently set to 10. If the condition is still true after 10 iterations, the step ends in a failure.
+
+The `parallel` step allows executing multiple branches (in parallel) specified in the `next_steps` field. At this time, the steps in `next_steps` are executed sequentially. Parallel execution is scheduled for a later release in SOARCA.
+
+Next, we explain variables in CACAO and SOARCA.
+
+### Variables
+
+The CACAO specification allows defining variables on the playbook level, as well as on the step level. Playbook variables are available throughout the playbook. Variables defined on the step level are available in that step, and in any step that executes in a sub-branch of an `if-condition`, `while-condition`, or `parallel` step.
```

Comment:
This is also matter of debate. At the moment, the specification does not define if-condition, while-condition, and parallel steps variables scope, and it seems you only have playbook-level or step-level variables. Spec text on variables scope for reference (10.18.1):
_The scope of a variable is determined by where the variable is declared. A variable may be defined globally for the entire playbook or locally within a workflow step. Variables are scoped to the object they are defined in, and any object that is used or referenced by that object._

Though, we could say that this is how we handle variables scope in SOARCA

Reply:
Indeed, the spec is not clear on this. My interpretation is that step variables are available within a step, and within steps that are spawned by `on_true`, `on_false`, and `next_steps` (i.e. execution branches if you want to call it that). I've also made my issues with the current wording of the `in_args` clear.